### ISSUE TITLE
API to refresh the server root with props

### ISSUE THREAD

**Issue by shuding** [vercel/next.js#32332]
To provide a way for client components to re-render the server component root from the client, with updated props, we need a new API `unstable_useRefreshRoot`. It works similar to the "location" thing in the very first RSC demo:

```js
// foo.client.js

const refresh = unstable_useRefreshRoot()

refresh({ ...props })
```

where `props` is a serializable object. When calling `refresh`, the RSC root will be re-rendered with these additional `props`. Also this API isn't allowed to be used on the server side.

As a note, we will need to move the implementation of this API into `ServerRoot` after #30994 lands.

**Comment by balazsorban44** [comment]
This issue has been automatically locked due to no recent activity. If you are running into a similar issue, please create a new issue with the steps to reproduce. Thank you.
